src/atlantis/router/backend/server.go

```diff
@@ -78,9 +78,10 @@ func (s *Server) Handle(logRecord *logger.HAProxyLogRecord, tout time.Duration)
 	logRecord.UpdateTr(tstart, tend)
 	select {
 	case resErr := <-resErrCh:
-		if resErr.Error == nil {
+		if resErr.Response != nil {
 			defer resErr.Response.Body.Close()
-
```

Comment:
This error is ultimately coming from the built-in [RoundTrip](http://golang.org/pkg/net/http/#RoundTripper) function.  The assumption was that if the error was not nil (the other branch, below), we don't have to worry about closing the Response.Body, since there isn't a response.  However, [this post](http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/) notes:

> Most of the time when your http request fails the resp variable will be nil and the err variable will be non-nil. However, when you get a redirection failure both variables will be non-nil. This means you can still end up with a leak.

I don't know if that applies here, but in any case, it's safer to defer the body close any time there's a response, regardless of what should or shouldn't happen.